src/V3Width.cpp

```diff
@@ -4000,9 +4033,20 @@ class WidthVisitor final : public VNVisitor {
             }
         }
         if (nodep->timingControlp()) {
-            nodep->timingControlp()->v3warn(
-                ASSIGNDLY, "Unsupported: Ignoring timing control on this assignment.");
-            nodep->timingControlp()->unlinkFrBackWithNext()->deleteTree();
+            if (v3Global.opt.timing()) {
+                if (VN_IS(m_ftaskp, Func)) {
+                    nodep->timingControlp()->v3error(
+                        "Timing controls are not legal in functions. Suggest use a task "
+                        "(IEEE 1800-2017 13.4.4)");
+                    VL_DO_DANGLING(nodep->timingControlp()->unlinkFrBack(), nodep);
+                    return;
+                }
```

Comment:
As above, illegal code should not be accepted even without timing enabled

Reply:
Fixed